src/shared/tpm2-util.c

```diff
@@ -2333,3 +2350,20 @@ int pcr_mask_to_string(uint32_t mask, char **ret) {
         *ret = TAKE_PTR(buf);
         return 0;
 }
+
+void tpm2_util_pbkdf(const void *key,
+                    size_t key_size,
+                    const void *input,
+                    size_t input_size,
+                    int iterations,
+                    uint8_t res[static SHA256_DIGEST_SIZE]) {
+
+        const void *salt = input;
+        size_t salt_size = input_size;
```

Comment:
maybe add `assert(iterations > 0)` here? because otherwise this will calculate garbage, no?